app/models/driver_adapter.rb

```diff
@@ -9,20 +10,33 @@ class DriverAdapter
 
   class << self
     def connection
-      ActiveRecord::Base.using(:grossman).connection
+      Octopus.using(:grossman) do
```

Comment:
We discussed refactoring these common methods into a base class. Something like: 

```ruby
GrossmanAdapter < ActiveRecord::Base
  class << self
    def connection
      Octopus.using(:grossman) do
        ActiveRecord::Base.connection
      end
    end

    def connected_with_octopus?
      Octopus.using(:grossman) do
        ActiveRecord::Base.connected_with_octopus?
      end
    end

    def not_supported
      fail 'Not supported!'
    end
    alias :new :not_supported
    alias :delete :not_supported
    alias :destroy :not_supported
  end
end
```

Also, can't you remove the `Octopus.using(:grossman) do` by using a `super.using(:grossman)` at the top of the class?

At which point you could completely drop the connection method from self.

So, in theory:

```ruby
# grossman_adapter.rb
GrossmanAdapter < ActiveRecord::Base
  self.using(:grossman)

  class << self
    def not_supported
      fail 'Not supported!'
    end
    alias :new :not_supported
  end

  def not_supported
    GrossmanAdapter.not_supported
  end
  alias :delete, :not_supported
  alias :destroy, :not_supported

  def sanitize_sql(sql)
    ActiveRecord::Base.send(:sanitize_sql_array, sql)
  end
end

# driver_adapter.rb
class DriverAdapter < GrossmanAdapter
  class << self
    def all
      results = connection.execute(sanitize_sql([QUERY, 'DR']))
      results.map(&:to_snake_keys)
    end
  end
end
```

I think you might want to look at the alias as some of those methods would be instance, some class.

_The alias example are guess work, not sure if they would work._